v0/destinations/delighted/util.js

```diff
@@ -0,0 +1,97 @@
+
+const axios = require("axios");
+const {
+    CustomError,
+    getValueFromMessage
+} = require("../../util");
+const { ENDPOINT } = require("./config");
+
+
+function isValidEmail(email){
+    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(String(email).toLowerCase());
+}
+function isValidPhone(phone){
+    var phoneformat= /^\+[1-9]\d{10,14}$/;
+    return phoneformat.test(String(phone));
+}
+
+const channelValidity = (channel, userId) => {
+    if(channel === "email"){
+        if(!isValidEmail(userId)){
+            throw new CustomError(
+                "Email format is not correct.",
+                400
+            );
+        }
+    }else if(channel === "phone"){
+        if(!isValidPhone(userId)){
+            throw new CustomError(
+                "Phone number format must be E.164.",
+                400
+            );
+        }
+    }else{
+        throw new CustomError(
+            "User Id is not matching the channel type.",
+            400
+        );
+    }
+    return channel;
+};
+
+const userValidity = async (channel , Config) => {
+    let getpayload = {};
+    if(channel === "email"){
+        getpayload.email = userId;
+    }else if(channel === "phone"){
+        getpayload.phone = userId;
+    }else{
+        throw new CustomError(
+            "Unable to generate payload for GET request.",
+            400
+        );
+    }
+    let response ;
+    try{
+        response = await axios.get(`${ENDPOINT}`,getpayload,{
+            headers:{
+                "Authorization" : `Basic ${Config.apiKey}`,
+                "Content-Type" : "application/json"
+            }
+        });
+        if(response && response.status === 200 ){
+            if(Array.isArray(response.data) && response.data.length === 0 )
+            return false;
+        return true;
+        }
+    }catch(error){
+        throw new CustomError("Error occured while searching user", JSON.stringify(error.response.data));
+    }
+}
+const  eventValidity = (Config, message) => {
```

Comment:
can we refactor this function?

Reply:
yeah , like instead of creating list and then comparing we can check while extracting. is this okay?